torch/distributed/rpc/api.py

```diff
@@ -73,27 +95,28 @@ def wait_all_workers():
         >>> # do some work
         >>> result = rpc.rpc_sync("worker1", torch.add, args=(torch.ones(1), 1))
         >>> # ready to shutdown
-        >>> rpc.wait_all_workers()
+        >>> rpc.shutdown()
 
         On worker 1:
         >>> import torch.distributed.rpc as rpc
         >>> rpc.init_rpc("worker1", rank=1, world_size=2)
         >>> # wait for worker 0 to finish work, and then shutdown.
-        >>> rpc.wait_all_workers()
+        >>> rpc.shutdown()
     """
     global _agent
-
     if _agent:
-        _agent.join()
-        _agent = None
+        if graceful:
+            _wait_all_workers()
         _destroy_rref_context(_ignore_rref_leak)
-        # clean up python rpc handler in wait_all_workers(), see comments in
+        _agent.shutdown()
+        # clean up python rpc handler in shutdown(), see comments in
         # PythonRpcHandler::cleanup(), call it in python API because the
         # cleanup() function has python dependency, it assumes python
         # interpreter exists
         _cleanup_python_rpc_handler()
+        _agent = None
```

Comment:
cc @mrshenli this is the new shutdown sequence after the ignore rref leaks PR has landed

Reply:
this looks good to me, thanks!